### ISSUE TITLE
TestEnv NPE in disposeSingleTool

### ISSUE THREAD

**Issue by astrelsky** [NationalSecurityAgency/ghidra#2288]
**Describe the bug**
A NPE is thrown during the teardown of AbstractProgramBasedTest due to the tool being null in the ProgramManagerPlugin.

**To Reproduce**
1. Run an AbstractProgramBasedTest?

**Attachments**
<details><summary>console log</summary>

```
Java HotSpot(TM) 64-Bit Server VM warning: Archived non-system classes are disabled because the java.system.class.loader property is specified (value = "ghidra.GhidraClassLoader"). To use archived non-system classes, this property must not be set
INFO  Using log config file: jar:file:/C:/Users/astre/Documents/ghidra_9.2_DEV/Ghidra/Framework/Generic/lib/Generic.jar!/generic.log4j.xml (LoggingInitialization)
INFO  Using log file: C:\Users\astre\AppData\Local\Temp\DocumentsTemp\application.log (LoggingInitialization)
INFO  Loading user preferences: C:\Users\astre\AppData\Local\Temp\DocumentsTemp\preferences (Preferences)
INFO  Class search complete (2491 ms) (ClassSearcher)
INFO  Initializing SSL Context (SSLContextInitializer)
INFO  Initializing Random Number Generator... (SecureRandomFactory)
INFO  Random Number Generator initialization complete: SHA1PRNG (SecureRandomFactory)
INFO  Trust manager disabled, cacerts have not been set (ApplicationTrustManagerFactory)
INFO  Test timeout feature disabled (ConcurrentTestExceptionStatement)
INFO  Creating project: C:\Users\astre\AppData\Local\Temp\DocumentsTemp\Ghidra-Cpp-Class-Analyzer_DevTestProject (DefaultProject)
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender (file:/C:/Users/astre/Documents/ghidra_9.2_DEV/Ghidra/Features/Base/lib/org.apache.felix.framework-6.0.3.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
INFO  Opened program in TestCodeBrowser tool: /TestProgram (ProgramManagerPlugin)
ERROR Unexpected exception closing tool: TestCodeBrowser (TestEnv) java.lang.NullPointerException: Cannot invoke "ghidra.framework.plugintool.PluginTool.contextChanged(docking.ComponentProvider)" because "this.tool" is null
	at ghidra.app.plugin.core.progmgr.ProgramManagerPlugin.updateActions(ProgramManagerPlugin.java:671)
 ProgramManagerPlugin.java
	at ghidra.app.plugin.core.progmgr.ProgramManagerPlugin.saveData(ProgramManagerPlugin.java:506)
 ProgramManagerPlugin.java
	at ghidra.framework.plugintool.PluginManager.saveData(PluginManager.java:631)
 PluginManager.java
	at ghidra.framework.plugintool.PluginTool.close(PluginTool.java:1128)
 PluginTool.java
	at ghidra.test.TestEnv.disposeSingleTool(TestEnv.java:272)
 TestEnv.java
	at ghidra.test.TestEnv.lambda$dipsoseTestTools$7(TestEnv.java:253)
 TestEnv.java
	at generic.test.AbstractGenericTest.runSwing(AbstractGenericTest.java:1149)
 AbstractGenericTest.java
	at generic.test.AbstractGenericTest.runSwing(AbstractGenericTest.java:1102)
 AbstractGenericTest.java
	at ghidra.test.TestEnv.dipsoseTestTools(TestEnv.java:252)
 TestEnv.java
	at ghidra.test.TestEnv.lambda$disposeTestTools$22(TestEnv.java:1137)
 TestEnv.java
	at generic.test.AbstractGenericTest.lambda$runSwing$8(AbstractGenericTest.java:1161)
 AbstractGenericTest.java
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
 InvocationEvent.java
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
 EventQueue.java
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
 EventQueue.java
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
 EventQueue.java
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
 AccessController.java
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
 ProtectionDomain.java
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
 EventQueue.java
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
 EventDispatchThread.java
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
 EventDispatchThread.java
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
 EventDispatchThread.java
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
 EventDispatchThread.java
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
 EventDispatchThread.java
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
 EventDispatchThread.java
```
</details>

**Environment (please complete the following information):**
 - OS: Microsoft Windows [Version 10.0.19041.508]
 - Java Version: 15
 - Ghidra Version: 9.2_DEV 02e017f50754d8ce8c91fa63b8adb931b4ec7796

**Additional context**
Just in case this is what my initialize looks like. Also, I'm getting another AutoAnalysisManager later before retreiving an Analyzer so I'm not bothering to get another after disposing it. The builder is a custom ProgramBuilder and it's init method is just laying down everything in memory for the test.
```java
@Override
protected void initialize() throws Exception {
	super.initialize();
	env.addPlugin(ClassTypeInfoManagerPlugin.class);
	AutoAnalysisManager man = AutoAnalysisManager.getAnalysisManager(program);
	// dispose it so the available analyzers are refreshed
	man.dispose();
	builder.init();
}
```

**Comment by dragonmacher** [comment]
I am not seeing this in master.  Are you overriding the `tearDown()` method?   It appears as thought the tool is being disposed twice.

**Comment by astrelsky** [comment]
> I am not seeing this in master. Are you overriding the `tearDown()` method? It appears as thought the tool is being disposed twice.

I am not overriding `tearDown()`

**Comment by dragonmacher** [comment]
So either dispose is being called twice, or the setup never actually initialized the plugin correctly.    Debugging the issue is the most direct way to see what is happening.   There may be some exception happening in an unexpected place that is causing this side-effect.

**Comment by astrelsky** [comment]
I found the problem. It's a transaction issue in ProgramBuilder. I'll put in a pr to fix it.

**Comment by astrelsky** [comment]
Fixed by 4629568298bc9dd3ec9b60e713de5170a4c53f6c
